src/main/scala/org/broadinstitute/dsde/firecloud/model/ESMapping.scala

```diff
@@ -0,0 +1,12 @@
+package org.broadinstitute.dsde.firecloud.model
+
+case class AttributeDefinition(properties: Map[String, AttributeDetail])
+case class AttributeDetail(`type`: String, items: Option[AttributeDetail]=None)
+
+case class ESMapping(mappings: ESDataset)
+case class ESDataset(dataset: ESDatasetProperty)
+case class ESDatasetProperty(properties: Map[String, Either[ESDetail, ESArray]])
```

Comment:
I'm confused about the need for `ESDataset` here. Could `ESMapping` just contain `properties: Map[String, Either[ESDetail, ESArray]]`?

Further: if you had a common parent for ESDetail and ESArray, something like:
```
sealed trait ESProperty
case class ESArray(properties: ESItem) extends ESProperty
case class ESDetail(`type`: String) extends ESProperty
````

then you could get rid of the use of `Either`, and just have a `Map[String, ESProperty]`.

Finally: we may not need the `ESArray` type at all. See https://www.elastic.co/guide/en/elasticsearch/reference/current/array.html ... I don't think that ES needs to be told that the field is an array.